eth/contracts/XPoap.sol

```diff
@@ -0,0 +1,262 @@
+pragma solidity ^0.5.0;
+
+import "zos-lib/contracts/Initializable.sol";
+import "openzeppelin-eth/contracts/token/ERC721/ERC721.sol";
+import "openzeppelin-eth/contracts/token/ERC721/ERC721Enumerable.sol";
+import "openzeppelin-eth/contracts/token/ERC721/IERC721Metadata.sol";
+import "./PoapRoles.sol";
+import "./PoapPausable.sol";
+
+
+    // Desired Features
+    // - Add Event
+    // - Add Event Organizer
+    // - Mint token for an event
+    // - Batch Mint
+    // - Burn Tokens (only admin?)
+    // - Pause contract (only admin)
+    // - ERC721 full interface (base, metadata, enumerable)
+
+contract XPoap is Initializable, ERC721, ERC721Enumerable, PoapRoles, PoapPausable {
+    event EventToken(uint256 eventId, uint256 tokenId);
```

Comment:
Hey @alavarello checkout this suggestion: https://github.com/poapxyz/poap/pull/10/files